index.js

```diff
@@ -7,7 +7,7 @@ var EventEmitter = require('events').EventEmitter
 var Request = require('./lib/request')
 var url = require('url')
 var defaults = {
-    port: 80,
+    port: (protocol) => protocol == 'https:' ? 443 : 80,
```

Comment:
sorry, this looks confusing/asymmetric to me. Can we avoid functions where we only need a constant mapping? How about something like this (i'm open to other ideas)? Alternatively, we can try to get the changes from the upstream repo (http-browserify), but that might be a much bigger project
```js
var defaultProtocol = 'http:'
var defaultPortByProtocol = {
  'http:': 80,
  'https:': 443,
}

// parse protocol first
// then parse port
```

Reply:
Let me think about it. I'm a bit busy today but I can put some time into it tomorrow.

I'm more than happy to take a look into getting changes from http-browserify as well. Your project is used by a lot of people so I think making sure it's as solid as possible is worth the extra work.

Reply:
Hey Tradle. I took a look into http-browserify and that project hasn't had an update in 7 years. I looked for implementations or tests from other popular projects but I couldn't find something applicable.

What do you think about refactoring the logic and adding some tests?

Reply:
I am new to react-native.

My aim is to create a LAN http server in react-native. How can I achive this ?

If I install [@tradle/react-native-http][4] and run . but, it occurs an error like below :

    var Stream = require('stream');
           stream module not found.

I have one doubt is it possible to run the below http code with this plugin in react-native.

    var http = require('http');

    //create a server object:
    http.createServer(function (req, res) {
       res.write('Hello World!'); //write a response to the client
       res.end(); //end the response
    }).listen(8080); //the server object listens on port 8080 

this plugin not worked , So, I tried another technic.

I want to use **http Built-in Module** in react-native. so , I followed [this][5] tutorial to **Use Core Node JS Modules in React Native Apps**

But, an error occured

I just followed [this][1] technic to use Core Node JS Modules in React Native Apps.

**Project structure**

[![enter image description here][2]][2]

**My code :**

    import {AppRegistry} from 'react-native';
    import App from './App';
    import {name as appName} from './app.json';
    import crypto from './crypto-in';
    import http from './http-in';
    import fs from './fs-in';


    // Create an instance of the http server to handle HTTP requests
    let app = http.createServer((req, res) => {
    // Set a response type of plain text for the response
    res.writeHead(200, {'Content-Type': 'text/html'});
    fs.createReadStream('index.html').pipe(res)
    });

    // Start the server on port 3000
    app.listen(3000, '127.0.0.1');
    console.log('Node server running on port 3000');
  
    const secret = 'open-sesame';
    const hash = crypto.createHmac('sha256', secret)
                   .update('abcdefg')
                   .digest('hex');
    console.log("*********************************** crypto = "+hash+" 
    ***************************************");

    AppRegistry.registerComponent(appName, () => App);


If I run it comes an error like this :

     BUNDLE  ./index.js

     WARN  The provided value 'ms-stream' is not a valid 'responseType'.
     WARN  The provided value 'moz-chunked-arraybuffer' is not a valid 'responseType'.
     ERROR  TypeError: undefined is not a function (near '..._httpIn.default.createServer...')
     ERROR  Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect.
      This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native.
    ERROR  Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect.
    This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native.


Why this error comes ? 

How can I create a LAN http server in react-native ?

  [1]: https://medium.com/hackernoon/using-core-node-js-modules-in-react-native-apps-64acd4d07140
  [2]: https://i.stack.imgur.com/HtIJK.png
  [4]: https://www.npmjs.com/package/@kierajs/react-native-http
  [5]: https://medium.com/hackernoon/using-core-node-js-modules-in-react-native-apps-64acd4d07140

Reply:
@Jeevasimba try [rn-nodeify](https://github.com/tradle/rn-nodeify)

Reply:
> @Jeevasimba try [rn-nodeify](https://github.com/tradle/rn-nodeify)


I try to add rn-nodeify in my react-native project with the help of this [doc][1].

But , I have some errors like.
    
    Error with react-native-os module 

How can I use rn-nodeify in my project ?

What is the purpose of rn-nodeify to use in react-native project ?

please explain !!!

  [1]: https://github.com/tradle/rn-nodeify/blob/master/readme.md



How can I use this module ?

Reply:
[This is my project.][1]

I add [npm install @tradle/react-native-http][2] in my project.

But , I have this error.

[![enter image description here][3]][3]

     [Error: undefined Unable to resolve module url from C:\Users\mahima\Desktop\react- 
    native\httpmob\node_modules\@tradle\react-native-http\index.js: url could not be found within the project or in these directories:
      node_modules
      ..\..\..\node_modules
       6 | var EventEmitter = require('events').EventEmitter
       7 | var Request = require('./lib/request')
    >  8 | var url = require('url')
         |                    ^
       9 | var defaults = {
       10 |     port: 80,
       11 |     protocol: 'http:']
     Error: Unable to resolve module url from C:\Users\mahima\Desktop\react- 
     native\httpmob\node_modules\@tradle\react-native-http\index.js: url could not be found within the project or in these directories:
      node_modules
      ..\..\..\node_modules
       6 | var EventEmitter = require('events').EventEmitter
       7 | var Request = require('./lib/request')
    >  8 | var url = require('url')
         |                    ^
       9 | var defaults = {
      10 |     port: 80,
      11 |     protocol: 'http:'


How can I solve this error ?

  [1]: https://github.com/Jeevasimba/httpmob
  [2]: https://github.com/tradle/react-native-http
  [3]: https://i.stack.imgur.com/tLryT.png